### PR TITLE
[kie-tools#2571] [sonataflow-management-console] Convert Containerfiles to Cekit

### LINKED ISSUE #2571: [sonataflow-management-console] Convert Containerfiles to Cekit
See: https://github.com/apache/incubator-kie-issues/issues/1470

The new SonataFlow Management Console introduces a new image to kie-tools. Since we already have the Cekit context enabled in the repository, we can easily convert this image to Cekit modules and leverage cache and BDD tests for this artifact.

### PR DESCRIPTION
**Pull request by ricardozanini**
Fixes #2571 

In this PR:

- Fixed a bug in the `sonata flow-image-common` `kogito-system-user` when a base image already has the 1001 user id.
- Removed the `Containerfile` in favor of Cekit to support reproducible builds and BDD tests in the future
- Upgrade Cekit and docker-squash dependencies
- Changed to the Red Hat httpd24 image based on ubi9 instead of manually installing and configuring the package. The httpd is tailored for this specific use case and is updated occasionally to fix CVEs in the httpd package. So, we avoid bringing this responsibility to this repo

Follow-up issues:

- https://github.com/apache/incubator-kie-tools/issues/2578

### CODE REVIEW

File: packages/sonataflow-management-console-image/package.json [hunk 1]
```diff
@@ -15,6 +15,10 @@
   "scripts": {
     "build:dev": "pnpm cleanup && pnpm env-json:schema:generate && pnpm copy:assets && run-script-if --bool \"$(build-env containerImages.build)\" --then \"pnpm image:docker:build\"",
     "build:prod": "pnpm cleanup && pnpm env-json:schema:generate && pnpm copy:assets && run-script-if --bool \"$(build-env containerImages.build)\" --then \"pnpm image:docker:build\"",
+    "cekit:build": "pnpm cleanup && pnpm env-json:schema:generate && pnpm copy:assets && cekit:copy && pnpm cekit:setup:env && make -C ./dist-dev build",
+    "cekit:copy": "run-script-os",
+    "cekit:copy:linux:darwin": "rimraf dist-dev && cp -R ./node_modules/@kie-tools/sonataflow-image-common/resources ./dist-dev/ && cp -R resources/* ./dist-dev/",
```
Comment (fantonangeli) [inline packages/sonataflow-management-console-image/package.json]: Do we miss `cekit:copy:linux:win32`, even just an `echo` for consistency?
ie: https://github.com/apache/incubator-kie-tools/blob/7b1cb0392c725b02263317b2db60404ab953f431/packages/serverless-logic-web-tools-base-builder-image/package.json#L19

File: packages/sonataflow-management-console-image/package.json [hunk 2]
```diff
@@ -15,6 +15,10 @@
   "scripts": {
     "build:dev": "pnpm cleanup && pnpm env-json:schema:generate && pnpm copy:assets && run-script-if --bool \"$(build-env containerImages.build)\" --then \"pnpm image:docker:build\"",
     "build:prod": "pnpm cleanup && pnpm env-json:schema:generate && pnpm copy:assets && run-script-if --bool \"$(build-env containerImages.build)\" --then \"pnpm image:docker:build\"",
+    "cekit:build": "pnpm cleanup && pnpm env-json:schema:generate && pnpm copy:assets && cekit:copy && pnpm cekit:setup:env && make -C ./dist-dev build",
```
Comment (fantonangeli) [inline packages/sonataflow-management-console-image/package.json]: make was not receiving env vars
```suggestion
    "cekit:build": "pnpm cleanup && pnpm env-json:schema:generate && pnpm copy:assets && pnpm cekit:copy && pnpm cekit:setup:env make -C ./dist-dev build",
```

File: packages/sonataflow-management-console-image/package.json (hunk 1 above)
Comment (fantonangeli) [inline packages/sonataflow-management-console-image/package.json]: `rimraf` was deleting the build of the previous commands, and the first `cp -R` was copying a dir inside `dist-dev`
```suggestion
    "cekit:copy:linux:darwin": "cp -R ./node_modules/@kie-tools/sonataflow-image-common/resources/* ./dist-dev/ && cp -R resources/* ./dist-dev/",
```

File: packages/sonataflow-management-console-image/resources/modules/sonataflow/management-console/configure-httpd.sh [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+SOURCES_DIR=/tmp/artifacts
+SCRIPT_DIR=$(dirname "${0}")
+MGMT_CONSOLE_HOME="${KOGITO_HOME}/management-console"
+
+# Configure the default httpd conf
+echo "Mutex posixsem" >> "${HTTPD_MAIN_CONF_PATH}/httpd.conf"
+sed -i -e "/#ServerName www.example.com:80/aHeader set Content-Security-Policy \"frame-ancestors 'self';\"" "${HTTPD_MAIN_CONF_PATH}/httpd.conf"
+sed -i -e 's/Options Indexes FollowSymLinks/Options -Indexes +FollowSymLinks/' "${HTTPD_MAIN_CONF_PATH}/httpd.conf"
+sed -i "s/Listen 80/Listen ${KOGITO_MANAGEMENT_CONSOLE_PORT}/g" "${HTTPD_MAIN_CONF_PATH}/httpd.conf"
+sed -i "s/#ServerName www.example.com:80/ServerName 127.0.0.1:${KOGITO_MANAGEMENT_CONSOLE_PORT}/g" "${HTTPD_MAIN_CONF_PATH}/httpd.conf"
```
Comment (fantonangeli) [inline packages/sonataflow-management-console-image/resources/modules/sonataflow/management-console/configure-httpd.sh]: in this package we use `SONATAFLOW_MANAGEMENT_CONSOLE_PORT` instead `KOGITO_MANAGEMENT_CONSOLE_PORT`
```suggestion
sed -i "s/Listen 80/Listen ${SONATAFLOW_MANAGEMENT_CONSOLE_PORT}/g" "${HTTPD_MAIN_CONF_PATH}/httpd.conf"
sed -i "s/#ServerName www.example.com:80/ServerName 127.0.0.1:${SONATAFLOW_MANAGEMENT_CONSOLE_PORT}/g" "${HTTPD_MAIN_CONF_PATH}/httpd.conf"
```

File: packages/sonataflow-management-console-image/resources/incubator-kie-sonataflow-management-console-image.yaml [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    - name: "maintainer"
+      value: "Apache KIE <dev@kie.apache.org>"
+    - name: "io.k8s.description"
+      value: "SonataFlow Management Console Image."
+    - name: "io.k8s.display-name"
+      value: "Sonataflow Management Console"
+    - name: "io.openshift.tags"
+      value: "sonataflow,serverless,workflow"
+    - name: "io.openshift.expose-services"
+      value: "$KOGITO_MANAGEMENT_CONSOLE_PORT:http"
```
Comment (fantonangeli) [inline packages/sonataflow-management-console-image/resources/incubator-kie-sonataflow-management-console-image.yaml]:
```suggestion
      value: "$SONATAFLOW_MANAGEMENT_CONSOLE_PORT:http"
```

File: packages/sonataflow-management-console-image/resources/incubator-kie-sonataflow-management-console-image.yaml [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    repositories:
+      - path: modules
+    install:
+      - name: org.kie.kogito.system.user
+      - name: org.kie.kogito.logging
+      - name: org.kie.kogito.project.versions
+      - name: org.kie.sonataflow.management.console
+
+  ports:
+    - value: $KOGITO_MANAGEMENT_CONSOLE_PORT
```
Comment (fantonangeli) [inline packages/sonataflow-management-console-image/resources/incubator-kie-sonataflow-management-console-image.yaml]: it doesn't seems to me that it really takes the env var here. If I hard code `8080` I don't have the message:
`["Value 'SONATAFLOW_MANAGEMENT_CONSOLE_PORT' is not of type 'int'. Path: '/value'"]`
 
```suggestion
    - value: $SONATAFLOW_MANAGEMENT_CONSOLE_PORT
```

File: packages/sonataflow-management-console-image/resources/modules/sonataflow/management-console/module.yaml [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+schema_version: 1
+name: org.kie.sonataflow.management.console
+version: "0.0.0"
+description: "SonataFlow Management Console Launch Scripts and Configurations"
+
+execute:
+  - script: configure-httpd.sh
+
+artifacts:
+  - name: management-console-app
```
Comment (fantonangeli) [inline packages/sonataflow-management-console-image/resources/modules/sonataflow/management-console/module.yaml]: I get this error but IDK what should be the path:
`2024-09-11 13:19:01,133 resource.py:229        INFO  Copying resource 'management-console-app'...
2024-09-11 13:19:01,133 resource.py:254        WARNING Cekit is not able to fetch resource 'management-console-app' automatically. Please use cekit-cache command to add this artifact manually.
2024-09-11 13:19:01,134 cli.py:544        ERROR Error copying resource: 'management-console-app'. See logs for more info.`

**Review by fantonangeli — COMMENTED** [review]
@ricardozanini Thank you so much for this PR.
I left some suggestions but I'm not totally sure about all of them.
I think we also need to rename `KOGITO_MANAGEMENT_CONSOLE_PORT` to `SONATAFLOW_MANAGEMENT_CONSOLE_PORT` in the `Containerfile`.
Of course, if this renaming makes sense for you as well.

File: packages/sonataflow-management-console-image/resources/modules/sonataflow/management-console/module.yaml (hunk 6 above)
Comment (fantonangeli) [reply packages/sonataflow-management-console-image/resources/modules/sonataflow/management-console/module.yaml]: I think cekit only accept zip files.
Locally I managed to pass through this step (just to try) creating `sonataflow-management-console-webapp.zip` with the webapp inside and using this:
```yaml
    path: ./sonataflow-management-console-webapp.zip
    md5: d6e88c463796b9ccd2bfaa92e188a3f0
```

File: packages/sonataflow-management-console-image/package.json (hunk 1 above)
Comment (ricardozanini) [reply packages/sonataflow-management-console-image/package.json]: This is a WIP, the final package.js won't be like that.

**Comment by ricardozanini** [comment]
@fantonangeli, thanks for the insights!

**Review by fantonangeli — APPROVED** [review]
LGTM, Thanks a lot @ricardozanini

**Comment by ricardozanini** [comment]
@fantonangeli where u able to build and run the image locally?

**Comment by fantonangeli** [comment]
@ricardozanini Yes I tested successfully the image. I used the commands:
- `pnpm cekit:build`
- `docker run --rm -it -p 8080:8080 docker.io/apache/incubator-kie-sonataflow-management-console:main`
Is that correct?

**Review by fantonangeli — APPROVED** [review]
@ricardozanini Re-tested with the last commits.

**Review by tiagobento — APPROVED** [review]

**Pull request merged**